pkg/flowaggregator/clickhouseclient/clickhouseclient.go

```diff
@@ -0,0 +1,511 @@
+}
+
+func (ci *ClickHouseInput) getDataSourceName() (string, error) {
+	if len(ci.DbURL) == 0 || len(ci.Username) == 0 || len(ci.Password) == 0 {
+		return "", fmt.Errorf("URL, Username or Password missing for clickhouse DSN")
+	}
+	var sb strings.Builder
+	sb.WriteString(fmt.Sprintf("%s?username=%s&password=%s", ci.DbURL, ci.Username, ci.Password))
+
+	if len(ci.Database) > 0 {
+		sb.WriteString("&database=")
+		sb.WriteString(ci.Database)
+	}
+	if ci.Debug {
+		sb.WriteString("&debug=true")
+	}
+	if ci.Compress {
+		sb.WriteString("&compress=true")
+	}
+
+	return sb.String(), nil
+}
+
+type ClickHouseFlowRow struct {
+	flowStartSeconds                     time.Time
+	flowEndSeconds                       time.Time
+	flowEndSecondsFromSourceNode         time.Time
+	flowEndSecondsFromDestinationNode    time.Time
+	flowEndReason                        uint8
+	sourceIP                             string
+	destinationIP                        string
+	sourceTransportPort                  uint16
+	destinationTransportPort             uint16
+	protocolIdentifier                   uint8
+	packetTotalCount                     uint64
+	octetTotalCount                      uint64
+	packetDeltaCount                     uint64
+	octetDeltaCount                      uint64
+	reversePacketTotalCount              uint64
+	reverseOctetTotalCount               uint64
+	reversePacketDeltaCount              uint64
+	reverseOctetDeltaCount               uint64
+	sourcePodName                        string
+	sourcePodNamespace                   string
+	sourceNodeName                       string
+	destinationPodName                   string
+	destinationPodNamespace              string
+	destinationNodeName                  string
+	destinationClusterIP                 string
+	destinationServicePort               uint16
+	destinationServicePortName           string
+	ingressNetworkPolicyName             string
+	ingressNetworkPolicyNamespace        string
+	ingressNetworkPolicyRuleName         string
+	ingressNetworkPolicyRuleAction       uint8
+	ingressNetworkPolicyType             uint8
+	egressNetworkPolicyName              string
+	egressNetworkPolicyNamespace         string
+	egressNetworkPolicyRuleName          string
+	egressNetworkPolicyRuleAction        uint8
+	egressNetworkPolicyType              uint8
+	tcpState                             string
+	flowType                             uint8
+	sourcePodLabels                      string
+	destinationPodLabels                 string
+	throughput                           uint64
+	reverseThroughput                    uint64
+	throughputFromSourceNode             uint64
+	throughputFromDestinationNode        uint64
+	reverseThroughputFromSourceNode      uint64
+	reverseThroughputFromDestinationNode uint64
+}
+
+func NewClickHouseClient(input ClickHouseInput) (*ClickHouseExportProcess, error) {
+	dsn, err := input.getDataSourceName()
+	if err != nil {
+		klog.Errorf("error parsing ClickHouse DSN: %v", err)
```

Comment:
```suggestion
		klog.ErrorS(err, "Error parsing ClickHouse DSN")
```
Could you switch to structured logging for all logs added in this patch? and log should be capitalized.